src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java

```diff
@@ -423,9 +434,33 @@ public InnReachResponseDTO finalCheckIn(String trackingId, String centralCode, B
       .ifPresent(this::removeHoldingsTransactionInfo);
     clearPatronAndItemInfo(transaction.getHold());
 
+    executeDeleteVirtualRecordsWithDelay(transaction.getHold());
+
     return success();
   }
 
+  private void executeDeleteVirtualRecordsWithDelay(TransactionHold hold) {
+
+    var folioItemId = hold.getFolioItemId();
+    var folioHoldingId = hold.getFolioHoldingId();
+    var folioInstanceId = hold.getFolioInstanceId();
+    var folioLoanId = hold.getFolioLoanId();
+
+    var configDataList=
+            configurationService.fetchConfigurationsDetailsByModule(CHECKOUT);
+
+    Long checkOutTimeDuration = getCheckOutTimeDuration(configDataList.getResult());
+
+    log.info("Checkout Time Duration is : {}", checkOutTimeDuration);
+
+    log.info("deleteVirtualRecords execution started at : " + new Date());
```

Comment:
Think this log message could be improved by containing both the current time and the time the execution is being scheduled for. In other words something like:

`log.info("deleteVirtualRecords execution scheduled at {} to run after a delay of  {} seconds as specified in the configuration value of checkOutTimeDuration", new Date(), checkoutTimeDuration)`

What is the unit for `checkoutTimeDuration` (minutes, seconds, days)? It would be good to see that documented someplace in the code.

Reply:
I didn't notice that you have `checkoutTimeDuration` in the line above. Still maybe would be better to combine the two log messages, and we should have the unit of `checkoutTimeDuration` (seconds?) specified in the log message and hopefully elsewhere.

Reply:
suggestions applied